app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -2131,6 +2132,22 @@ class BrowserTabFragment :
         webShareRequest.launch(JsCallbackData(data, featureName, method, id))
     }
 
+    private fun permissionsQuery(featureName: String, method: String, id: String, data: JSONObject) {
+        val url = runBlocking(dispatchers.main()) {
+            webView?.url
+        }
```

Comment:
Why do we need this?
`runBlocking` inside an imperative function feels odd. We're doing something odd here.

Reply:
Agreed, overall this is kinda weird, it als calls the web chrome client when that's not needed. Instead of that I'd move the logic from the WebChromeClient to the TabViewModel, there you can check the URL by accessing `site.url` rather than jumping on the main thread. The view model can then return a command containing the jscallbackdata so contentScopeScripts can call onResponse with it.